responsibilities.md

```diff
@@ -0,0 +1,75 @@
+# Software development lifecycle
+
+## Requirements review and handoff
+
+* Coordinate with @Diggy (Product) and @Tyler (Operations) first
+* Identify product owner
+* Identify any security considerations or blockers with @mrnerdhair
+* Get a public meeting booked on the DAO calendar - use the `#request-calendar-events` channel in Discord
+* Invite community contributors to kickoff meetings!
+* Document technical requirements in this docs repo - create a new section
+
+## Architecture planning and review
+
+* Book times with @Major Hayes or @0xdef1cafe
```

Comment:
Not sure if we are there yet, but eventually it may be good to have a standing slot for design reviews. I think it's really helpful for socializing our architecture decisions and more knowledge sharing.